src/mozanalysis/experiment.py

```diff
@@ -531,6 +532,43 @@ def _build_enrollments_query_fenix_baseline(self, time_limits):
             dataset=self.app_id or "org_mozilla_firefox",
         )
 
+    def _build_enrollments_query_glean_baseline(self, time_limits, dataset):
+        """Return SQL to query enrollments for a Glean no-event experiment
+
+        If enrollment events are available for this experiment, then you
+        can take a better approach than this method. But in the absence
+        of enrollment events (e.g. in a Mako-based experiment, which
+        does not send enrollment events), you need to fall back to using
+        ``ping_info.experiments`` to get a list of who is in what branch
+        and when they enrolled.
+        """
+        # Try to ignore users who enrolled early - but only consider a
+        # 7 day window
+        return """
+            SELECT events.client_info.client_id AS client_id,
+                mozfun.map.get_key(
+                    e.extra,
+                    'branch'
+                ) AS branch,
+                DATE(MIN(events.submission_timestamp)) AS enrollment_date
+            FROM `moz-fx-data-shared-prod.{dataset}.events` events,
+            UNNEST(events.events) AS e
+            WHERE
+                DATE(events.submission_timestamp)
+                BETWEEN DATE_SUB('{first_enrollment_date}', INTERVAL 7 DAY)
```

Comment:
in line with comment above

```suggestion
                BETWEEN '{first_enrollment_date}'
```